File: src/gestorAplicacion/personal/Empleado.java
package gestorAplicacion.personal;

import java.util.ArrayList;
import java.util.List;
import gestorAplicacion.tienda.*;
/**
 * 
 * @author Esteban García
 * @summary Clase padre de Tecnico y Dependiente. Define los comportamientos en comun y en general de las clases que heredan de ella.
 * Punto importante: La lista servicios, la cual mantiene todos los servicios en los que un empleado esté trabajando/involucrado.
 *
 */
public abstract class Empleado implements Personal {

	private String nombre;
	private int cedula;
	protected List<Servicio> servicios;
	private double cartera;
	private static List<Empleado> empleados;
	
	public Empleado (String nombre, int cedula) {
		this.nombre = nombre;
		this.cedula = cedula;
		servicios = new ArrayList<Servicio>();
		Empleado.empleados.add(this);
	}

	public String getNombre() {
		return nombre;
	}

	public void setNombre(String nombre) {
		this.nombre = nombre;
	}

	public int getCedula() {
		return cedula;
	}

	public void setCedula(int cedula) {
		this.cedula = cedula;
	}

	public List<Servicio> getServicios(){
		return servicios;
	}

	public void setServicios(List<Servicio> servicios) {
		this.servicios = servicios;
	}
	
	
	
	
}

File: practica2/generic_it/gestor_aplicacion/personal/empleado.py
class Empleado:
    _empleados = []

    def __init__(self, nombre, cedula):
        self._nombre = nombre
        self._cedula = cedula
        self._servicios = []
        self._cartera = 0
        Empleado._empleados.append(self)
    
    def getNombre(self):
        return self._nombre
    
    def setNombre(self, nombre):
        self._nombre = nombre
    
    def getCedula(self):
        return self._cedula
    
    def setCedula(self, cedula):
        self._cedula = cedula
    
    def getServicios(self):
        return self._servicios
    
    def setServicios(self, servicios):
        self._servicios = servicios
    
    def getCartera(self):
        return self._cartera
    
    def setCartera(self, cartera):
        self._cartera = cartera
    
    @classmethod
    def getEmpleados(cls):
        return cls._empleados
    
    @classmethod
    def setEmpleados(cls, empleados):
        cls._empleados = empleados
    
    def __str__():
        pass
